src/Microsoft.AspNetCore.Razor.Language/Legacy/HtmlMarkupParser.cs

```diff
@@ -537,6 +543,125 @@ private bool BangTag()
             return false;
         }
 
+        private HtmlSymbol AcceptAllButLastDoubleHypens()
+        {
+            var lastDoubleHyphen = CurrentSymbol;
+            AcceptWhile(s =>
+            {
+                if (NextIs(HtmlSymbolType.DoubleHyphen))
+                {
+                    lastDoubleHyphen = s;
+                    return true;
+                }
+
+                NextToken();
+                EnsureCurrent();
+                return false;
+            });
+
+            if (At(HtmlSymbolType.Text) && IsDashSymbol(CurrentSymbol))
```

Comment:
Not sure I follow the purpose of this block, could you explain?

Reply:
Think about the following case: "--->" = "--" + "->" = DoubleHyphen + "-" + ClosingAngle
The tokenizer will start parsing the long dashes from the beginning, accepting doubleHyphen symbols.
First symbol will be processed, and the leftover part will be "->". This, by itself isn't really a comment closing. As that logic is purely based on our approach, we should make sure we process that case in swapped order. So here I approach it:
"--->" = "--" + "-" + ">" = "-" + "--" + ">"

So here I don't accept the last double hyphen symbol but keep it in memory just in case there'll be a single dash there. If so, I first accept the dash, so I have double hyphen and closing angle nicely stacked up. Hope this makes sense

Reply:
In the case of a triple dash, doesnt the last `-` get re-ordered?